components/api_version/src/api_v2.rs

```diff
@@ -168,6 +169,42 @@ impl KvFormat for ApiV2 {
             encoded_key
         }
     }
+
+    // add prefix RAW_KEY_PREFIX
+    fn convert_encoded_key_version_from(
+        src_api: ApiVersion,
+        key: &[u8],
+        ts: Option<TimeStamp>,
+    ) -> Result<Key> {
+        match src_api {
+            ApiVersion::V1 | ApiVersion::V1ttl => {
+                let mut apiv2_key = Vec::with_capacity(ApiV2::get_encode_len(key.len() + 1));
+                apiv2_key.push(RAW_KEY_PREFIX);
+                apiv2_key.extend(key);
+                Ok(Self::encode_raw_key_owned(apiv2_key, ts))
+            }
+            ApiVersion::V2 => Ok(Key::from_encoded_slice(key)),
```

Comment:
For the whole PR, there is a question, should we check the validation of the key/range while converting them into or from `ApiV2`

Reply:
From BR side, we should check the validition, but only can check it when backup from apiv2, and it's checked in `backup::utils::is_valid_raw_value`. If it's from apiv1/apiv1ttl, it's all valid as user cannot use Raw and Txn in one same cluster. From api_version side, i think it's hard the check the range validition, like the range end key 'r' + 1. So the consensus we reach is that the user of api_version(BR/storage) should guarantee the input's range validition.